### DISCUSSION TITLE
widget in View action

### DISCUSSION THREAD

**Discussion by ghost** [filamentphp/filament#6610] [Help]
Is that possible to create a widget in record view action? I mean when I click to View, modal opens and in this modal I would like to add a widget.

**Comment by pboivin** [comment]
Hi @onexcloud, I don't think it's possible to include widgets in the ViewAction's modal but you may be able to work around it with a custom ViewField : https://filamentphp.com/docs/2.x/forms/fields#view

You could set it to be visible only in the 'view' context :
```
    ViewField::make('my_view_field')
        ->view('my-custom-view')
        ->visibleOn('view'),
```

**Reply by danharrin** [reply]
You can pass a view to the ->modalContent() method of the action, then you dont need form fields for something that isnt a form
